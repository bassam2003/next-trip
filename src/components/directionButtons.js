import  React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { Paper, Button } from '@material-ui/core'
import { header } from '../utils/styleConstants'

const styles = {
  root: { 
    margin: '20px', 
    display: 'flex', 
    flexDirection: 'row', 
    alignItems: 'center' 
  },
  button: {
    margin: '10px',
  },
  header,
}

export const DirectionButtons = ({ 
    classes = {},
    directions = [], 
    currentDirection = '', 
    onSelectDirection = () => {} 
  }) => {
    return (
    <Paper style={{ textAlign: 'center' }}>
    <h1 className={classes.header}>Direction</h1>
    <div className={classes.root}>
      {directions.map((direction, idx) => 
        <Button 
          key={idx}
          className={classes.button}
          variant="contained" 
          color={currentDirection === direction.Value ? "primary" : ''}
          onClick={() => onSelectDirection(direction.Value)}
        >
          {direction.Text}
        </Button>
      )}
    </div>
    </Paper>
  )
  }

export default withStyles(styles)(DirectionButtons)