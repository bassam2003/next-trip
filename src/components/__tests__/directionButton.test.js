import React from 'react'
import { shallow } from 'enzyme'
import { DirectionButtons } from '../directionButtons'

describe('DirectionButtons', () => {
  it('should render without error', () => {
    const wrapper = shallow(<DirectionButtons />)

    expect(wrapper).toMatchSnapshot()
  })
})