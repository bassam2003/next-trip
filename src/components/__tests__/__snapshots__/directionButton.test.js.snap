// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`DirectionButtons should render without error 1`] = `ShallowWrapper {}`;
